modules/tftp/tftp_api.rb

```diff
@@ -18,8 +18,9 @@ def instantiate(variant, mac = nil)
         Object.const_get("Proxy").const_get('TFTP').const_get(variant.capitalize).new
       end
 
-      def create(variant, mac)
+      def create(variant, mac, os, major, minor, arch, bootfilename_efi, build)
```

Comment:
Given how many arguments there are, I'd consider using keyword arguments
```suggestion
      def create(variant, mac, os: nil, major: nil, minor: nil, arch: nil, bootfilename_efi: nil, build: nil)
```